File: frontend/Md.Tga.Terminator/application/services/pub-sub.js
const { PubSub } = require('@google-cloud/pubsub');
const uuid = require('uuid');

const pubSubClient = new PubSub();

/**
 * Initializes a pub/sub client.
 * @param {object} options An options object.
 * @param {string} options.topicName The name of the pub/sub topic.
 * @returns A new pub/sub client.
 */
const initialize = (options) => {
  const { topicName } = options;

  const client = {
    /**
     * Publish a message to pub/sub.
     * @param {object} message The message that is sent.
     * @param {string} message.gameSeriesId The id of the game series.
     * @param {string} message.gameId The id of the game.
     * @param {Array} message.terminationId The termination id of the player.
     * @param {Array} message.winningSideId The side that should win the game.
     */
    publish: async (message) => {
      const {
        gameSeriesId,
        gameId,
        terminationId,
        winningSideId,
        reason,
      } = message;

      const json = `{"processId":"${uuid.v4()}","gameSeriesId":"${gameSeriesId}","gameId":"${gameId}","terminationId":"${terminationId}","winningSideId":"${winningSideId}","reason":"${reason}"}`;

      const data = Buffer.from(json);
      await pubSubClient
        .topic(topicName)
        .publishMessage({ data });
    },
  };

  return client;
};

module.exports = initialize;

File: backend/Md.Tga.IntegrationTest/pub_sub.py
'''
Module pub_sub provides operations for sending google cloud pub/sub messages.
'''
from google.cloud import pubsub_v1

class PubSubClient:
  '''
  Client for sending google cloud pub/sub messages.
  '''

  def __init__(self, project_id: str):
    '''
    Initializes a new instance of the PubSubClient class.

    Args:
        project_id (str): The id of the project to that messages are sent.
    '''
    self.__project_id = project_id
    self.__pub_sub_client = pubsub_v1.PublisherClient()

  def publish(self, topic_id: str, message: str) -> None:
    '''
    Sends the message to the specifies topic.

    Args:
        topic_id (str): The id of the pub/sub topic.
        message (str): A message in json format.
    '''
    topic = self.__pub_sub_client.topic_path(self.__project_id, topic_id)
    response = self.__pub_sub_client.publish(topic, message.encode())
    response.result()
